packages/vkui/src/components/CalendarHeader/CalendarHeader.test.tsx

```diff
@@ -47,4 +49,68 @@ describe('CalendarHeader', () => {
       container.getElementsByClassName(styles['CalendarHeader__nav-icon-next'])[0],
     ).not.toBeTruthy();
   });
+  it('do not display prev and next month buttons if isMonthDisabled return true', () => {
+    const onChange = jest.fn();
+    const { container } = render(
+      <CalendarHeader viewDate={targetDate} onChange={onChange} isMonthDisabled={() => true} />,
+    );
+
+    expect(
+      container.getElementsByClassName(styles['CalendarHeader__nav-icon-prev'])[0],
+    ).not.toBeTruthy();
+
+    expect(
+      container.getElementsByClassName(styles['CalendarHeader__nav-icon-next'])[0],
+    ).not.toBeTruthy();
+  });
+  it('do not display prev month button when set prevMonthHidden prop', () => {
+    const onChange = jest.fn();
+    const { container } = render(
+      <CalendarHeader viewDate={targetDate} onChange={onChange} prevMonthHidden />,
+    );
+
+    expect(
+      container.getElementsByClassName(styles['CalendarHeader__nav-icon-prev'])[0],
+    ).not.toBeTruthy();
+  });
+  it('do not display next month button when set nextMonthHidden prop', () => {
+    const onChange = jest.fn();
+    const { container } = render(
+      <CalendarHeader viewDate={targetDate} onChange={onChange} nextMonthHidden />,
+    );
+
+    expect(
+      container.getElementsByClassName(styles['CalendarHeader__nav-icon-next'])[0],
+    ).not.toBeTruthy();
+  });
+
+  it('does not fire onChange when click on month dropdown item if isMonthDisabled return true', async () => {
+    const onChange = jest.fn();
+    jest.useFakeTimers();
+    const { container } = render(
+      <CalendarHeader viewDate={targetDate} isMonthDisabled={() => true} onChange={onChange} />,
+    );
+
+    const [monthPicker] = container.getElementsByClassName(styles['CalendarHeader__picker']);
+    await userEvent.click(monthPicker);
+    const [januarySelectOption] = container.getElementsByClassName(styles['CustomSelectOption']);
+    await userEvent.click(januarySelectOption);
+
+    expect(onChange).not.toHaveBeenCalled();
+  });
+
+  it('does not fire onChange when click on year dropdown item if isYearDisabled return true', async () => {
+    const onChange = jest.fn();
+    jest.useFakeTimers();
+    const { container } = render(
+      <CalendarHeader viewDate={targetDate} isYearDisabled={() => true} onChange={onChange} />,
+    );
+
+    const [, yearPicker] = container.getElementsByClassName(styles['CalendarHeader__picker']);
+    await userEvent.click(yearPicker);
+    const [minYearSelectOption] = container.getElementsByClassName(styles['CustomSelectOption']);
+    await userEvent.click(minYearSelectOption);
+
+    expect(onChange).not.toHaveBeenCalled();
+  });
```

Comment:
Тут я бы тоже не привязывался к селекторам. И постарался бы искать по тексту, при этом проверив, что это работает без `isYearDisabled`.

```suggestion
  it('does not fire onChange when click on year dropdown item if isYearDisabled return true', async () => {
    const onChange = jest.fn();
    jest.useFakeTimers();
    const component = render(
      <CalendarHeader viewDate={targetDate} onChange={onChange} changeYearLabel="Изменить год" />,
    );

    await userEvent.click(screen.getByLabelText('Изменить год'));
    await userEvent.click(screen.getByText('2020'));
    expect(onChange).toHaveBeenCalled();

    onChange.mockReset();

    component.rerender(
      <CalendarHeader
        isYearDisabled={() => true}
        viewDate={targetDate}
        onChange={onChange}
        changeYearLabel="Изменить год"
      />,
    );

    await userEvent.click(screen.getByLabelText('Изменить год'));
    await userEvent.click(screen.getByText('2020'));

    expect(onChange).not.toHaveBeenCalled();
  });

```

Reply:
Подумаю. Поиск по тексту мне показался не очень удачным, т.к. он завязан на локаль.

Reply:
Да, есть такое дело, лучше по testid но тут мы не даём их прокидывать, хотя могли бы, и даже стоило бы, опять же, чтобы пользователи библиотеки могли не звисеть от локали и приватных классов в e2e тестах.

Reply:
Добавлю отдельным PR.